save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -34,12 +37,72 @@ data class Warning(
     "SwallowedException",
     "ThrowsCount")
 internal fun String.extractWarning(warningRegex: Regex,
+                                   fileName: String?,
+                                   lineGroupIdx: Int?,
                                    columnGroupIdx: Int?,
+                                   messageGroupIdx: Int,
+): Warning? {
+    val groups = warningRegex.find(this)?.groups ?: return null
+
+    val line = lineGroupIdx?.let {
+        try {
+            groups[lineGroupIdx]!!.value.toInt()
+        } catch (e: Exception) {
+            throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.message}")
+        }
+    }
+
+    val column = columnGroupIdx?.let {
+        try {
+            groups[columnGroupIdx]!!.value.toInt()
+        } catch (e: Exception) {
+            throw ResourceFormatException("Could not extract column number from line [$this], cause: ${e.message}")
+        }
+    }
+
+    val message = try {
+        groups[messageGroupIdx]!!.value
+    } catch (e: Exception) {
+        throw ResourceFormatException("Could not extract warning message from line [$this], cause: ${e.message}")
+    }
+
+    return Warning(
+        message,
+        line,
+        column,
+        fileName,
+    )
+}
+
+/**
+ * Extract warning from [this] string using provided parameters
+ *
+ * @param warningRegex regular expression for warning
+ * @param columnGroupIdx index of capture group for column number
+ * @param lineGroupIdx index of capture group for line number
+ * @param messageGroupIdx index of capture group for waring text
+ * @param fileNameGroupIdx index of capture group for file name
+ * @return a [Warning] or null if [this] string doesn't match [warningRegex]
+ * @throws ResourceFormatException when parsing a file
+ */
+@Suppress(
+    "TooGenericExceptionCaught",
+    "SwallowedException",
+    "ThrowsCount")
+internal fun String.extractWarning(warningRegex: Regex,
```

Comment:
This method could simply extract the file name and then call the overloaded version. Now you have a lot of copy-pasted code

Reply:
Fixed